src/components/tiptap/notificationWidget/NotificationWidget.tsx

```diff
@@ -101,10 +102,17 @@ const NotificationComponent = ({
   name: string
   route: PortalRoutes
 }) => {
+  const appState = useAppState()
+  const pathname = usePathname()
   return (
     <Stack direction='row' justifyContent='space-between'>
       <Typography variant='body1'>{name}</Typography>
-      <RedirectButton route={route}>
+      <RedirectButton
+        route={route}
+        execute={
+          pathname.includes('client-preview') || !appState?.appState.readOnly
+        }
```

Comment:
Shouldn't `pathname.includes('client-preview')` alone be sufficient?

Reply:
`!appState?.appState.readOnly` is necessary to tackle the `readonly` case in preview mode. @rrojan